src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/aws/AWSEC2JCloudsProvider.java

```diff
@@ -277,28 +285,34 @@ private void addOptions(Template template, Options options, Infrastructure infra
 
     }
 
-    private void allowPortInSecurityGroup(int port, String securityGroup, String region,
-            SecurityGroupApi securityGroupApi) {
+    private void allowPortInSecurityGroup(int port, String securityGroupId, String region,
+            AWSSecurityGroupApi awsSecurityGroupApi) {
         if (port == -1) {
```

Comment:
can you add a comment here explaining what's the significance of a negative port

Reply:
I still don't know @alijawadfahs. It was already there, I did not change it to keep the same behavior. 
Your comment is not related to my PR.

Reply:
Here the port -1 means "all ICMP/ICMPv6 types"
According to the doc: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_IpPermission.html#API_IpPermission_Contents
`A value of -1 indicates all ICMP/ICMPv6 types.`

@mklkun could you help to add this into a comment here please? If you haven't yet merged the PR. Thank you.

Reply:
Thanks @tao-xinxiu for the explanation. Sure I will add that!